Ec explain (#904)
Dasharo/docs: docs/transparent-validation/ec-information/ec-explained.md

> # EC power states in NovaCustom laptops

Reviewer: let's make this a bit more generic, I suspect we'll be adding more sections here sooner or later

> ## Way the EC works right now

Reviewer: ```suggestion
## Power states

This section describes the logic used by the EC to control platform power states
depending on AC adapter presence.
```

> ### Turning off the computer while charger is plugged in

Reviewer: ```suggestion
### Shutdown, with AC connected
```

The machine switches to system power state S5, using little to no
power. As a result pressing the power button to turn it on, will result in
a warmboot. (S0 → S5 → S0)

> ### Cutting off power feed while in S5

Reviewer: ```suggestion
### AC detach while turned off
```

EC disables all power planes, and the platform is set to G3 power state.
Therefore, in that case pressing the power button results in a coldboot.
(S0 → S5 → G3 → S0)

> ### Turning the computer off while not plugged in

Reviewer: ```suggestion
### Shutdown, with AC disconnected
```

> The machine actually enters S5 briefly and immediately after, it switches
> to G3 like intended. Like mentioned before, turning the computer from G3
> results in a coldboot. (S0 → S5 → G3 → S0 )

Reviewer: ```suggestion
The machine actually enters S5 and proceeds to G3 soon after. Pressing the
power button will result a coldboot. (S0 → S5 → G3 → S0 )
```
